contracts/src/v0.8/vrf/dev/VRFConsumerBaseV2PlusUpgradeable.sol

```diff
@@ -0,0 +1,185 @@
+ * @dev if your contract could have multiple requests in flight simultaneously.
+ *
+ * @dev Colliding `requestId`s are cryptographically impossible as long as seeds
+ * @dev differ.
+ *
+ * *****************************************************************************
+ * @dev SECURITY CONSIDERATIONS
+ *
+ * @dev A method with the ability to call your fulfillRandomness method directly
+ * @dev could spoof a VRF response with any random value, so it's critical that
+ * @dev it cannot be directly called by anything other than this base contract
+ * @dev (specifically, by the VRFConsumerBaseV2Plus.rawFulfillRandomness method).
+ *
+ * @dev For your users to trust that your contract's random behavior is free
+ * @dev from malicious interference, it's best if you can write it so that all
+ * @dev behaviors implied by a VRF response are executed *during* your
+ * @dev fulfillRandomness method. If your contract must store the response (or
+ * @dev anything derived from it) and use it later, you must ensure that any
+ * @dev user-significant behavior which depends on that stored value cannot be
+ * @dev manipulated by a subsequent VRF request.
+ *
+ * @dev Similarly, both miners and the VRF oracle itself have some influence
+ * @dev over the order in which VRF responses appear on the blockchain, so if
+ * @dev your contract could have multiple VRF requests in flight simultaneously,
+ * @dev you must ensure that the order in which the VRF responses arrive cannot
+ * @dev be used to manipulate your contract's user-significant behavior.
+ *
+ * @dev Since the block hash of the block which contains the requestRandomness
+ * @dev call is mixed into the input to the VRF *last*, a sufficiently powerful
+ * @dev miner could, in principle, fork the blockchain to evict the block
+ * @dev containing the request, forcing the request to be included in a
+ * @dev different block with a different hash, and therefore a different input
+ * @dev to the VRF. However, such an attack would incur a substantial economic
+ * @dev cost. This cost scales with the number of blocks the VRF oracle waits
+ * @dev until it calls responds to a request. It is for this reason that
+ * @dev that you can signal to an oracle you'd like them to wait longer before
+ * @dev responding to the request (however this is not enforced in the contract
+ * @dev and so remains effective only in the case of unmodified oracle software).
+ */
+
+import {Initializable} from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+
+/**
+ * @dev The VRFConsumerBaseV2PlusUpgradable is an upgradable variant of VRFConsumerBaseV2Plus
+ * @dev (see https://docs.openzeppelin.com/upgrades-plugins/1.x/writing-upgradeable).
+ * @dev It's semantics are identical to VRFConsumerBaseV2Plus and can be inherited from
+ * @dev to create an upgradeable VRF v2.5 consumer contract.
+ */
+
+abstract contract VRFConsumerBaseV2PlusUpgradable is IVRFMigratableConsumerV2Plus, ConfirmedOwner, Initializable {
```

Comment:
Since this is now an upgradable contract, inheriting from `ConfirmedOwner` doesn't make sense because it has a constructor. `ConfirmedOwner` can be removed, but that also means the possibility of ownership transfer becomes the responsibility of the consumer contract's implementation. Is that expected?

Reply:
Thanks for pointing out, @ibrajer.

Sure, I'll remove the inheritance of `ConfirmedOwner` then.

I need to set / initialize the owner in the `__VRFConsumerBaseV2Plus_init()` function and add the definition of `transferOwnership` within this abstract consumer contract itself.

Like:

```js
// solhint-disable-next-line func-name-mixedcase
function __VRFConsumerBaseV2Plus_init(address _vrfCoordinator, address _owner) internal onlyInitializing {
  if (_vrfCoordinator == address(0)  || _owner == address(0)) {
    // solhint-disable-next-line gas-custom-errors
    revert ZeroAddress();
  }

  s_vrfCoordinator = IVRFCoordinatorV2Plus(_vrfCoordinator);
  s_owner = _owner;
}

/// @notice Allows an owner to begin transferring ownership to a new address.
function transferOwnership(address to) public onlyOwner {
  s_owner = to;
}
```

So, as you've mentioned, the possibility of ownership transfer becomes the responsibility of the consumer contract's implementation.

Reply:
I would also suggest following a similar pattern as in the [ConfirmedOwnerWithProposal.sol](https://github.com/smartcontractkit/chainlink/blob/ef77fac68b4f6f7b55a869af9b1484ea8a98c845/contracts/src/v0.8/shared/access/ConfirmedOwnerWithProposal.sol).

This will ensure that no one will accidentally transfer ownership to a wrong or non-accessible account. By following this pattern, you will always require an extra step by the new owner: to accept the ownership of the consumer contract.